### ISSUE TITLE
Use role="tab" for mat-tab-link

### ISSUE THREAD

**Issue by calebegg** [angular/material2#11209]
Our accessibility reviewer says our tabs should have `role="tab"`, which makes sense to me. It seems like it's done for `mat-tab`, but not `mat-tab-link`.

**Comment by jelbourn** [comment]
The `role="tab"` interaction pattern is primarily for when the URL doesn't change. So when the URL _does_ does change, nav + anchors is more in line with expected semantics. Closing as working-as-intended.
